lib/mpl_toolkits/axes_grid1/parasite_axes.py

```diff
@@ -262,27 +240,29 @@ def twin(self, aux_trans=None, axes_class=None):
             parasite_axes_auxtrans_class_factory(axes_class)
 
         if aux_trans is None:
-            ax2 = parasite_axes_auxtrans_class(
-                self, mtransforms.IdentityTransform(), viewlim_mode="equal")
-        else:
-            ax2 = parasite_axes_auxtrans_class(
-                self, aux_trans, viewlim_mode="transform")
+            aux_trans = mtransforms.IdentityTransform()
+        ax2 = parasite_axes_auxtrans_class(
+            self, aux_trans, viewlim_mode="transform")
         self.parasites.append(ax2)
-        ax2._remove_method = self.parasites.remove
+        ax2._remove_method = self._remove_twinany
 
         self.axis["top", "right"].set_visible(False)
 
         ax2.axis["top", "right"].set_visible(True)
         ax2.axis["left", "bottom"].set_visible(False)
 
-        def _remove_method(h):
-            self.parasites.remove(h)
-            self.axis["top", "right"].set_visible(True)
-            self.axis["top", "right"].toggle(ticklabels=False, label=False)
-        ax2._remove_method = _remove_method
-
         return ax2
 
+    def _remove_twinany(self, ax):
```

Comment:
Maybe picky, but `remove_twinany` is hard to read. `remove_any_twin` or `remove_twins` would be clearer.

Reply:
any_twin it is.